src/tools/miri/src/shims/unix/dlsym.rs

```diff
@@ -46,8 +46,10 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
         match dlsym {
             Dlsym::Android(dlsym) =>
                 android::EvalContextExt::call_dlsym(this, dlsym, args, dest, ret),
+            #[allow(unreachable_code)]
             Dlsym::FreeBsd(dlsym) =>
                 freebsd::EvalContextExt::call_dlsym(this, dlsym, args, dest, ret),
+            #[allow(unreachable_code)]
             Dlsym::Linux(dlsym) => linux::EvalContextExt::call_dlsym(this, dlsym, args, dest, ret),
```

Comment:
same here? do we warn for unreachable code but you still can't remove the match arm or sth?

Reply:
Yes, removing the arm gives a non-exhaustive pattern error.

Reply:
i think it's a pretty big issue if the unreachable code lint has false positives. I guess we could instead write `match Dlsym {}` here. Or what's actually happening is that the fact that `Dlsym` is uninhabited is conceptionally private to the linux shim.

I think the fix should be to instead change `Dlsym` to

```rust
pub struct Dlsym(Void);
```